tests/test_optimistic_transaction.rs

```diff
@@ -67,3 +62,228 @@ pub fn test_optimistic_transaction() {
+    {
+        let mut opts = Options::default();
+
+        opts.create_if_missing(true);
+
+        let db = OptimisticTransactionDB::open(&opts, &path).unwrap();
+
+        let write_options = WriteOptions::default();
+
+        let optimistic_transaction_options = OptimisticTransactionOptions::new();
+
+        let trans1 = db.transaction(&write_options, &optimistic_transaction_options);
+
+        let mut optimistic_transaction_options_snapshot = OptimisticTransactionOptions::new();
+
+        optimistic_transaction_options_snapshot.set_snapshot(true);
+
+        // create transaction with snapshot
+        let trans2 = db.transaction(&write_options, &optimistic_transaction_options_snapshot);
+
+        trans1.put(b"k1", b"v1").unwrap();
+
+        let k1_2 = trans2.get(b"k1").unwrap();
+
+        assert!(k1_2.is_none());
+
+        trans1.commit().unwrap();
+
+        trans2.commit().unwrap();
+
+        drop(trans2);
+
+        let trans3 = db.transaction(&write_options, &optimistic_transaction_options_snapshot);
+
+        trans1.delete(b"k1").unwrap();
+
+        trans1.commit().unwrap();
+
+        let k1_3 = trans3.get(b"k1").unwrap().unwrap();
+
+        assert_eq!(&*k1_3, b"v1");
+
+        trans3.commit().unwrap();
+
+        drop(trans3);
+
+        let trans4 = db.transaction(&write_options, &optimistic_transaction_options_snapshot);
+
+        let k1_4 = trans4.get(b"k1").unwrap();
+
+        assert!(k1_4.is_none());
+
+        trans4.commit().unwrap();
+    }
+}
+
+#[test]
+pub fn test_optimistic_transaction_merge() {
+    fn concat_merge(
+        _new_key: &[u8],
+        existing_val: Option<&[u8]>,
+        operands: &mut MergeOperands,
+    ) -> Option<Vec<u8>> {
+        let mut result: Vec<u8> = Vec::with_capacity(operands.size_hint().0);
+        existing_val.map(|v| {
+            for e in v {
+                result.push(*e)
+            }
+        });
+        for op in operands {
+            for e in op {
+                result.push(*e)
+            }
+        }
+        Some(result)
+    }
+
+    let path = DBPath::new("_rust_rocksdb_optimistic_transaction_snapshot");
+
+    {
+        let mut opts = Options::default();
+
+        opts.create_if_missing(true);
+
+        opts.set_merge_operator("test operator", concat_merge, None);
+
+        let db = OptimisticTransactionDB::open(&opts, &path).unwrap();
+
+        let trans = db.transaction_default();
+
+        trans.put(b"k1", b"a").unwrap();
+
+        trans.merge(b"k1", b"b").unwrap();
+
+        trans.merge(b"k1", b"c").unwrap();
+
+        trans.merge(b"k1", b"d").unwrap();
+
+        trans.merge(b"k1", b"efg").unwrap();
+
+        let k1_result = trans.get(b"k1");
+
+        assert!(k1_result.is_err());
+
+        trans.commit().unwrap();
+
+        let k1 = trans.get(b"k1").unwrap().unwrap();
+
+        assert!(k1.to_utf8().unwrap() == "abcdefg");
+
+        trans.commit().unwrap();
```

Comment:
Could you please reformat these tests ? It looks a little bit strange as for me. I mean empty lines.

Reply:
ok

Reply:
ok？